pkg/apis/core/v1alpha1/federatedtypeconfig_types.go

```diff
@@ -236,6 +236,16 @@ func (f *FederatedTypeConfig) isPrimitiveNamespaced() bool {
 	return f.Spec.Namespaced
 }
 
+func (f *FederatedTypeConfig) GetFederatedKind() string {
+	// TODO(marun) Use the constant in pkg/controller/util
+	if f.Name == "namespaces" {
+		// The template type is 'Namespace', so return
+		// 'FederatedNamespace' for consistency with other types.
+		return "FederatedNamespace"
```

Comment:
It is used for messages. Make a "FederatedNamespace" seems a little confused. Because no one can get this as a resource in the cluster. "federated Namespace" seems better to me.

Reply:
As per the comment, `FederatedNamespace` is for consistency with the kind of every other type (e.g. FederatedDeployment, FederatedConfigMap, etc).  I don't think it makes sense to diverge from the kind form for namespaces.